### PR TITLE
Release preparations for 0.18

### PR DESCRIPTION
**Pull request by jku**
This is a draft for 0.18 release notes. Changelog should be fairly accurate
   *  it does not try to list all merges, but tries to list all interesting changes and the related merges
   * bug fixes for unreleased features are listed with the feature merge in the Added section, not in Fixed section, as that should make more sense to reader

I can't do the release builds as I don't have upload rights on pypi... so anything else come to mind before I hand this to you @joshuagl ?

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,53 @@
...
+  * mypy, black and isort integration (#1314, #1395, #1455, #1363)
+  * API reference documentation build (#1517)
+* Extensive Metadata API testing and validation (#1416, #1430, #1449, #1450,
+  #1451, #1460, #1466, #1511)
+* Add ngclient: a new client library implementation 
+  (#1408, #1448, #1463 #1467, #1470, #1474, #1501, #1509, #1524)
+
+### Removed
+* Remove Python 2 support (#1293)
+* Remove direct dependency on six
```
Comment (MVrachev) [inline docs/CHANGELOG.md:28]:
```suggestion
* Remove direct dependency on six (#1340)
```

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,53 @@
...
+* Remove obsolete reference to Thandy in a LICENSE file (#1472)
```
Comment (MVrachev) [inline docs/CHANGELOG.md:43]:
```suggestion
* Remove Python 2 support (#1293)
* Remove direct dependency on six (#1340)
* Remove direct dependency future (#1342)
* Remove obsolete reference to Thandy in a LICENSE file (#1472)
```

**Review by MVrachev — COMMENTED** [review]
I am not sure if it's ready for review, but some of the changes I see missing are:
#1291, #1299, #1328, #1359, #1367, #1372, #1386, #1394, #1415, #1477, #1489, #1522.
Maybe you consider some of them as non-important but have another look.
I tried to point out only the important ones according to me.

File: docs/CHANGELOG.md (hunk 2 above)
Comment (jku) [reply docs/CHANGELOG.md:43]: future is just a base python module (so I don't think deserves a mention)

**Comment by jku** [comment]
Last pushes:
* Added mention of ADR9
* Added mention of GH project name change

**Comment by jku** [comment]
last pushes:
* tweak language on stability status and recommended use for Metadata API and ngclient
* rebase on develop, add one more testing PR number to list

The author list for the release is (slightly massaged from `git shortlog -s  v0.17.0..HEAD | awk '{ print $2" "$3}' | sort`)

> hosseinsia 
> Joshua Lock
> Jussi Kukkonen
> Kainaat Singh
> Lukas Puehringer
> maoyangLiu 
> Marina Moore
> Martin Vrachev
> Philippe Coval
> samuelgregorovic 
> Teodora Sechkova
> Tim Gates
> Velichka Atanasova

I'm marking this non-draft and handing to @joshuagl .

**Review by joshuagl — APPROVED** [review]
LGTM, thanks for putting this together!

**Pull request merged**